kuma/javascript/src/article.jsx

```diff
@@ -89,8 +89,16 @@ function addAnchors(article) {
 
 export default function Article({ document }: DocumentProps) {
     const article = useRef(null);
-    useEffect(() => article.current && highlightSections(article.current));
-    useEffect(() => article.current && addAnchors(article.current));
+    useEffect(() => {
+        if (article.current) {
+            highlightSections(article.current);
+        }
+    }, [document]);
+    useEffect(() => {
+        if (article.current) {
+            addAnchors(article.current);
+        }
+    }, [document]);
```

Comment:
Wondering about the performance implications of comparing `document` after every render. Is there a smaller bit of data that could be compared to achieve the same result?

Reply:
It's not a deep comparison. Just checking if it's the same object. So dirt-cheap in terms of performance :)

Reply:
I think Peter is right about this, but the official documentation is unclear. But googling turns of lots of hits for people trying to make useEffect() do deep comparison, so I assume this really is just testing reference equality.

But thank you for bringing this up Jon. I should arguable be comparing the actual strings of html that the Article component displays instead of the object that contains them. I don't believe that our architecture ever allows the html to change without creating a new documentdata object, though so it is probably okay as it stands.

Reply:
Excellent 😁